src/configs/hardware-config.c

```diff
@@ -217,6 +217,18 @@ static void init_hardware_config(void) {
         wasm_v128_xor(xint8_output, wasm_i32x4_const_splat(-128)),
         wasm_v128_xor(overflow_output, wasm_i32x4_const(65536, 33024, 33024, 512))));
     }
+    {
+      // Check out-of-bounds behaviour of VNNI (vpdpbusd) version Relaxed Integer Dot Product with Accumulation.
```

Comment:
This works with (i8mm) vsudot on ARM as well?

Reply:
I'm not quite familiar with ARM. But I see a description [here](https://developer.arm.com/documentation/ddi0597/2024-03/SIMD-FP-Instructions/VSUDOT--by-element---Dot-Product-index-form-with-signed-and-unsigned-integers--by-element--?lang=en). It seem that this check may also work with vsudot, since vsudot deal with s8/u8 input and no saturation applied?

Reply:
Oh yes, vsudot would work identical to VNNI, if V8 supports that, but it is considered an I8MM instruction.  The more common instruction is vsdot which takes 2 signed int8.  So for ARM (or vnni-int8) we might like a version that does not add 128.